pkg/scheduler/framework/plugins/helper/node_affinity.go

```diff
@@ -68,3 +68,33 @@ func nodeMatchesNodeSelector(node *v1.Node, nodeSelector *v1.NodeSelector) bool
 	matches, _ := corev1.MatchNodeSelectorTerms(node, nodeSelector)
 	return matches
 }
+
+// GetPodRequiredNodeSelectorAndAffinity returns required node selector and affinity for the pod.
+func GetPodRequiredNodeSelectorAndAffinity(pod *v1.Pod) (labels.Selector, *v1.NodeSelector) {
+	var selector labels.Selector
+	if len(pod.Spec.NodeSelector) > 0 {
+		selector = labels.SelectorFromSet(pod.Spec.NodeSelector)
+	}
+	var affinity *v1.NodeSelector
+	if pod.Spec.Affinity != nil &&
+		pod.Spec.Affinity.NodeAffinity != nil {
+		affinity = pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution
+	}
+	return selector, affinity
```

Comment:
I suppose it's more performant to return `NewLazyErrorNodeSelector(affinity)` here; otherwise `NewLazyErrorNodeSelector(affinity)` will be called N times, via NodeMatchesNodeSelectorAndAffinity() -> MatchNodeSelectorTerms().

Reply:
Done.